### ISSUE TITLE
Horizontal Tag Alignment

### ISSUE THREAD

**Issue by jmayalag** [nicoespeon/gitgraph.js#132]
Tags aren't aligned correctly when orientation is horizontal. I don't know if this is the expected behavior.

![gitgraph](https://cloud.githubusercontent.com/assets/6710350/21400029/e4a60df8-c78c-11e6-889e-abf1b12aca6f.png)

```javascript
let gitflow = new GitGraph({
    template: "metro",
    orientation: "horizontal",
    mode: "compact",
  })

  let master = gitflow.branch({
    name: 'master',
    showLabel: true
  })

  master.commit({
    message: 'Initial',
    tag: 'v0.1.0'
  })

  let develop = master.branch({
    name: 'develop',
    showLabel: true,
  })

  develop.commit().commit()
  develop.merge(master)
  develop.commit()
  master.commit({
    message: 'Change',
    tag: 'v0.2.0'
  })
```

**Comment by cgddrd** [comment]
Having also come across this, I did a little digging. 

It looks like this behaviour is indeed intentional, having located the following call to `_drawTextBG`, that renders tag labels under 'horizontal mode' (line ????):

```
_drawTextBG( this.context, 
             this.x - this.dotSize / 2,
             ((this.parent.columnMax + 1) * this.template.commit.tag.spacingY) -  this.template.commit.tag.spacingY / 2 + (this.parent.tagNum % 2) * textHeight * 1.5,
             this.tag, 
             this.tagColor, 
             this.tagFont, 
             0, 
             this.displayTagBox );
```

Looking closely at the third parameter (Y position), we see the following: `(this.parent.tagNum % 2)`, which when multiplied by `textHeight * 1.5`, will cause every other tag label to be positioned higher than the previous. 

The following examples confirm these findings:

<img width="829" alt="screen shot 2016-12-27 at 15 51 43" src="https://cloud.githubusercontent.com/assets/2536819/21503079/75ccf5da-cc4c-11e6-9fd0-d9043290ea5d.png">

**Fig 1:** Graph rendered using existing behaviour (difference in Y position between even/odd tag labels)

<img width="817" alt="screen shot 2016-12-27 at 15 52 05" src="https://cloud.githubusercontent.com/assets/2536819/21503121/cb225502-cc4c-11e6-8f4e-ae44eaa7c7e8.png">

**Fig 2:** Graph rendered using modified behaviour (`2 + (this.parent.tagNum % 2) * textHeight * 1.5` modified to become: `2 + textHeight * 1.5`

If we can confirm with the original repo author that this is expected behaviour, and if so, whether it should remain so, then I would be happy to make any necessary modifications.

**Comment by nicoespeon** [comment]
Hi @jmayalag & @cgddrd!

First of all: happy new year 🎊 
I hope you've had great time with your family and friends 😄 

Thanks a lot for digging into this.
At first, this looked to me much more like an issue than an expected behavior. It's not intuitive why tags are not aligned.

However, after searching back in the history, I found that we actually did this to avoid tags collision which might happen if you tag two siblings commits: https://github.com/nicoespeon/gitgraph.js/commit/22b8f5fc9b675326f6722fc57e423184322a23b2

So this is "intended", but I think we can do better and actually check whether the offset is required so it will feel more natural.
